### ISSUE TITLE
Event received after being filtered in channel's handle_out

### ISSUE THREAD

**Issue by EskiMag** [phoenixframework/phoenix#3271]
### Environment

* Elixir version (elixir -v): 1.7.4
* Phoenix version (mix deps): 1.4.0 (56fe9a80)
* NodeJS version (node -v): n/a
* NPM version (npm -v): n/a
* Operating system: MacOS Mojave 10.14.3 (18D109)

### Expected behavior
When intercepting an event in my Phoenix Channel by function which is just returning `{:noreply, socket}`, I would expect the event to be filtered (not sent).

### Actual behavior
Event is sent to me even though I did not use the `push` function inside the `handle_out`.

> I hope I am not doing something incorrectly. Maybe we just need to update the docs. I would be happy to do that after clarification.

### Repo for reproduction
https://github.com/EskiMag/phoenix_handle_out

#### Relevant files:
* https://github.com/EskiMag/phoenix_handle_out/blob/master/lib/handle_out_web/channels/events_channel.ex
* https://github.com/EskiMag/phoenix_handle_out/blob/master/test/handle_out_web/channels/events_channel_test.exs

**Comment by Gazler** [comment]
Is this just an issue in the tests, or have you observed it in your application?

I believe this is intentional as assert/refute_broadcast is testing the pubsub and not your client. In your example, there is a broadcast for "event-out".

From the docs: https://hexdocs.pm/phoenix/Phoenix.ChannelTest.html#subscribe_and_join/4

 > By subscribing to the topic, we can use assert_broadcast/3 to verify a message has been sent through the pubsub layer.

I think this could improved in the documentation for assert/refute_broadcast

Something like

 > *NOTE* assert_broadcast/3 tests that the message was broadcast in the pubsub system. For testing if a client receives a message, use assert_push/3`

**Comment by EskiMag** [comment]
Thanks for clarification @Gazler. You're right, `refute_push` is returning `true`. It's just `refute_broadcast` which is returning `false`. I was confused because I saw `%Phoenix.Socket.Broadcast{event: "event-out", payload: %{}, topic: "events"}` in the mailbox of the socket process, which I inspected by calling `:erlang.process_info(self(), :messages) |> IO.inspect(label: "\n\nmailbox")`. I'll try to do the pull request for the docs update.
